src/com/github/emilbayes/downloadmanager/DownloadManagerPlugin.java

```diff
@@ -146,7 +146,7 @@ private static JSONArray JSONFromCursor(Cursor cursor) throws JSONException {
             rowObject.put("title", cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_TITLE)));
             rowObject.put("description", cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_DESCRIPTION)));
             rowObject.put("mediaType", cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_MEDIA_TYPE)));
-            rowObject.put("localFilename", cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_FILENAME)));
+            rowObject.put("localFilename", cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_URI)));
```

Comment:
Is this a mistake or on purpose?

Reply:
Nope, testing on android 7+ told me that `COLUMN_LOCAL_FILENAME` was deprecated and resulted to error in callback. Searching upon deprecation brought to solutions using `COLUMN_LOCAL_URI` instead.

Reply:
I don't think this is the right thing to do; https://developer.android.com/reference/android/app/DownloadManager#COLUMN_LOCAL_FILENAME

Reply:
I've seen this, I just thought that it was a bit overkilling for the purpose of the plugin. It's basically the same think just without the "file://" in front. So maybe just `DownloadManager.COLUMN_LOCAL_URI.replace("file://", "")`

Reply:
Ah yes, so the snippet they refer to I don't think belongs here. What I'm more considering is whether this property should be present if the API is > 24. That can however also quickly turn into a mess. What does other plugins do in cases like this? I know it's very common in iOS plugins to feature detect

Reply:
We could just use platform version detection for the purpose.

Reply:
Added feature detection in last commit